src/app/services/auth.service.ts

```diff
@@ -111,8 +116,12 @@ export class AuthService {
       );
   }
 
+  onStateChange() {
+    return this.isLogged$;
+  }
+
   setLocal(user: any) {
-    return this.localStorage.setItemSubscribe('user', user);
+    return this.localStorage.setItem('user', user).subscribe(ok => this.isLogged$.next('true'));
```

Comment:
aqui como você não usa essa variável `ok` pode deixar como
```suggestion
    return this.localStorage.setItem('user', user).subscribe(() => this.isLogged$.next('true'));
```